src/permutedims/permutedims.jl

```diff
@@ -28,29 +23,55 @@ function fusiontensor_permutedims(
   return fusiontensor_permutedims(ft, biperm)
 end
 
-function fusiontensor_permutedims(ft, biperm::BlockedPermutation{2})
+function fusiontensor_permutedims!(
+  ftdst, ftsrc, new_codomain_indices::Tuple, new_domain_indices::Tuple
+)
+  biperm = permmortar((new_codomain_indices, new_domain_indices))
+  return fusiontensor_permutedims!(ftdst, ftsrc, biperm)
+end
+
+# permutedims with BlockedPermutation
+function fusiontensor_permutedims(ft, biperm::AbstractBlockPermutation{2})
   ndims(ft) == length(biperm) || throw(ArgumentError("Invalid permutation length"))
+  ftdst = FusionTensor{eltype(ft)}(undef, axes(ft)[biperm])
+  fusiontensor_permutedims!(ftdst, ft, biperm)
+  return ftdst
+end
+
+function fusiontensor_permutedims!(ftdst, ftsrc, biperm::AbstractBlockPermutation{2})
+  ndims(ftsrc) == length(biperm) || throw(ArgumentError("Invalid permutation length"))
+  blocklengths(axes(ftdst)) == blocklengths(biperm) ||
+    throw(ArgumentError("Destination tensor does not match bipermutation"))
+  checkspaces(axes(ftdst), axes(ftsrc)[biperm])
 
-  # early return for identity operation. Do not copy. Also handle tricky 0-dim case.
-  if ndims_codomain(ft) == first(blocklengths(biperm))  # compile time
-    if Tuple(biperm) == ntuple(identity, ndims(ft))
-      return ft
+  # early return for identity operation. Also handle tricky 0-dim case.
+  if ndims_codomain(ftdst) == ndims_codomain(ftsrc)  # compile time
+    if Tuple(biperm) == ntuple(identity, ndims(ftdst))
+      copy!(data_matrix(ftdst), data_matrix(ftsrc))
+      return nothing
     end
   end
+  return permute_data!(SymmetryStyle(ftdst), ftdst, ftsrc, Tuple(biperm))
```

Comment:
```suggestion
  permute_data!(SymmetryStyle(ftdst), ftdst, ftsrc, Tuple(biperm))
  return ftdst
```
In general I like to use the convention that in-place function return the object being modified.